lighthouse-core/computed/metrics/lantern-speed-index.js

```diff
@@ -134,14 +134,12 @@ class LanternSpeedIndex extends LanternMetric {
       }
     }
 
-    if (!layoutWeights.length) {
-      return fcpTimeInMs;
-    }
-
     const totalWeightedTime = layoutWeights
       .map(evt => evt.weight * Math.max(evt.time, fcpTimeInMs))
       .reduce((a, b) => a + b, 0);
     const totalWeight = layoutWeights.map(evt => evt.weight).reduce((a, b) => a + b, 0);
+
+    if (!totalWeight) return fcpTimeInMs;
     return totalWeightedTime / totalWeight;
```

Comment:
it's hard to get my brain around possible values here without looking at a bunch of examples, but is it possible to get a weird discontinuity where adding a single layout task could give a result *lower* than `fcpTimeInMs`?

IOW, should this line be `return Math.max(totalWeightedTime / totalWeight, fcpTimeInMs);` or is that implicitly handled?

(this isn't something new, handling the division by zero case is 👍 )

Reply:
> is it possible to get a weird discontinuity where adding a single layout task could give a result lower than fcpTimeInMs

no the `.map(evt => evt.weight * Math.max(evt.time, fcpTimeInMs))` above should prevent any time earlier than `fcpTimeInMs` from being used in the weight

Reply:
Oh, yep, there it is :)